src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1718,6 +1718,52 @@ SPECTRE_ALWAYS_INLINE constexpr auto mutate_apply(
   mutate_apply(F{}, box, std::forward<Args>(args)...);
 }
 
+// @{
+/*!
+ * \ingroup DataBoxGroup
+ * \brief Perform a mutation to the \ref DataBoxGroup `box`, assigning the
+ * `args` to the tags `MutateTags` in order.
+ */
+template <typename... MutateTags, typename BoxTags, typename... Args>
```

Comment:
The `mutate_apply` function takes a `tmpl::list` and the `mutate` function takes a parameter pack of tags (not sure why). I suggest you use the `tmpl::list` approach for this function since I think it's a bit more flexible. What do you think?

Edit: I see you have an overload below that takes a list as a meta argument. I think the interface is better when the function takes the list as template parameter.

Reply:
I chose to keep an interface with a pack rather than a list so that it mirrored as much as possible the `db::mutate` function. Admittedly, it is a bit awkward that the list interface then needs to be a meta function argument.

If you don't think the analogy with the other utility is enough of a reason to keep this as-is, I can switch over to a pure-list interface.

Reply:
I see. `db::mutate` is actually inconsistent with `db::mutate_apply` (which does pretty much the same thing) and `db::apply`. @kidder could you weigh in here how you think this interface should look like, since you have been restructuring it recently?

Reply:
The main reason I think `db::mutate_assign` taking a typelist as template parameter would be good is that all the initialization actions use it with their `simple_tags` list. So the interface that takes a typelist is used a lot. You could make the interface mirror that of `db::mutate_apply` instead of `db::mutate` (see #2568).

Reply:
yes, I'll take a look

Reply:
I think making `db::mutate_assign` take a typelist to be consistent with `db::mutate_apply` makes sense.  I think `db::mutate` takes a pack to make it more convenient for mutating one tag

Reply:
ping on this thread. If I understand @kidder correctly, the interface both of us are suggesting is:

```cpp
template <typename MutateTags, typename BoxTags, typename... Args>
SPECTRE_ALWAYS_INLINE constexpr void mutate_assign(
     const gsl::not_null<DataBox<BoxTags>*> box, Args&&... args) noexcept;
```

This is consistent with `mutate_apply` in https://github.com/sxs-collaboration/spectre/pull/2568/.

Reply:
ping on this thread. If I understand @kidder correctly, the interface both of us are suggesting is:

```cpp
template <typename MutateTags, typename BoxTags, typename... Args>
SPECTRE_ALWAYS_INLINE constexpr void mutate_assign(
     const gsl::not_null<DataBox<BoxTags>*> box, Args&&... args) noexcept;
```

This is consistent with `mutate_apply` in https://github.com/sxs-collaboration/spectre/pull/2568/.